src/services/passport.ts

```diff
@@ -58,6 +60,105 @@ const initPassport = () => {
 			},
 		),
 	)
+
+	/**
+	 * Sign in with Google
+	 *
+	 * User has signed in
+	 *  - link user's account with Google if it has not been linked
+	 * User has not signed in
+	 *  - sign in user if google account has been used to sign up
+	 * 	- sign up user if google account has not been used to sign up
+	 */
+	passport.use(
+		new GoogleStrategy(
+			{
+				clientID: config.secrets.googleClientId,
+				clientSecret: config.secrets.googleClientSecret,
+				callbackURL: '/auth/google/callback',
+				passReqToCallback: true,
+			},
+			async (req, accessToken, refreshToken, profile, done) => {
+				const {
+					id: googleId,
+					name: {givenName, familyName},
+					emails: [{value: gmail}],
+				} = profile
+
+				try {
+					if (req.user) {
```

Comment:
If we do not use session, I wonder if this will ever be true?

Reply:
This case handles user if user has been logged in. So checkToken will add user object to req

Reply:
Rite